### ISSUE TITLE
[Bug]: Editing file in UI mode deletes its traces

### ISSUE THREAD

**Issue by gal-checksum** [microsoft/playwright#31155]
### Version

1.44.1

### Steps to reproduce

1. Launch Playwright UI Mode
2. Make sure watch mode is off
3. Run a test to view traces
4. Edit the test file
5. Save
6. Go back to view the trace of the test run

### Expected behavior

Even though you edited the test file, you can still see the traces of the latest test run.

### Actual behavior

Traces disappear and you must run the test again to view the test

### Additional context

This is useful incase you are editing a test in multiple places. You can use traces to update one action and then go back to the traces for the next one

### Environment

```shell
System:
    OS: macOS 14.2.1
    CPU: (8) arm64 Apple M1 Pro
    Memory: 103.67 MB / 16.00 GB
  Binaries:
    Node: 21.0.0 - ~/.nvm/versions/node/v21.0.0/bin/node
    Yarn: 1.22.19 - /usr/local/bin/yarn
    npm: 10.2.0 - ~/.nvm/versions/node/v21.0.0/bin/npm
    pnpm: 9.0.6 - /opt/homebrew/bin/pnpm
  IDEs:
    VSCode: 1.89.1 - /usr/local/bin/code
  Languages:
    Bash: 3.2.57 - /bin/bash
```

**Comment by mxschmitt** [comment]
I tried to reproduce it but was not able to. Does it happen for a [newly created project](https://playwright.dev/docs/intro)? How does your test file look like?

**Comment by gal-checksum** [comment]
It did when I opened the bug but now it doesn't. I'll try ti investigate more to understand under which conditions that might happen

**Comment by mxschmitt** [comment]
We had a similar bug in an older version of Playwright, but it should be fixed in recent versions like v1.44.1. Maybe its a different one or you used an older version?

**Comment by gal-checksum** [comment]
OK that's possible. I should have been on 1.44.1 but maybe I missed something. I'll update and let you know. Closing for now.
